### ISSUE TITLE
Rustdoc lists associated types in traits under "required methods"

### ISSUE THREAD

**Issue by huonw** [rust-lang/rust#20501]
They're not methods, so this is a little weird. e.g.

http://doc.rust-lang.org/std/ops/trait.Deref.html

> # Required Methods
> 
>     type Target
> 
>
>     fn deref(&'a self) -> &'a <Self as Deref>::Target
>
> The method called to dereference a value

**Comment by Gankro** [comment]
Should we just rename it to "Required Items"?

**Comment by nick29581** [comment]
It would be better to have associated items listed separately, I'd have thought. (I realise static methods are technically associated items, but I think from a usability perspective listing them separately is better).

**Comment by japaric** [comment]
This appears to be fixed as of `rustc 0.13.0-nightly (c7dd3c4d6 2015-01-05 23:51:00 +0000)`.

``` rust
#![no_std]
#![feature(associated_types, lang_items)]

#[lang = "sized"] pub trait Sized {}

pub trait Deref for Sized? {
    type Sized? Target;

    /// The method called to dereference a value
    fn deref<'a>(&'a self) -> &'a Self::Target;
}
```

![20150106123154](https://cloud.githubusercontent.com/assets/5018213/5632842/1c00935c-95a0-11e4-8052-fc4c2eb6b5d0.png)

But it seems that the fix hasn't made it to the official docs yet.

**Comment by nikomatsakis** [comment]
Closing as fixed, taking @japaric's word for it.
